### PR TITLE
add buildpulse config to sources

### PR DESCRIPTION
**Pull request by codevbus**
### Description:
Adds buildpulse integration config to the community test suite(sources).

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -66,5 +66,27 @@ jobs:
...
+    - name: Test 
+      run: |
+        CGO_ENABLED=0 go test -timeout=5m $(shell go list ./... | grep -v /vendor/ | grep -v pkg/detectors | grep -v 'pkg/sources')
+      if: ${{ success() || failure() }} # always run this step, even if there were previous errors
+    - name: Upload test results to BuildPulse for flaky test detection
+      if: '!cancelled()' # Run this step even when the tests fail. Skip if the workflow is cancelled.
+      uses: buildpulse/buildpulse-action@main
+      with:
+        account: 79229934
+        repository: 523156419
```
Comment (dustin-decker) [inline .github/workflows/test.yml:82]: Should be `repository: 77726177` I think

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -66,5 +66,27 @@ jobs:
...
       uses: actions/checkout@v3
-    - name: Test
-      run: make test-community
+    - name: Set up gotestsum
+      run: |
+        go install gotest.tools/gotestsum@latest
+        mkdir -p tmp/test-results
+    - name: Test 
+      run: |
+        CGO_ENABLED=0 go test -timeout=5m $(shell go list ./... | grep -v /vendor/ | grep -v pkg/detectors | grep -v 'pkg/sources')
```
Comment (dustin-decker) [inline .github/workflows/test.yml:75]: You need to run tests with gotestsum and if you provide the go test command to it then it must have json output

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -66,5 +66,27 @@ jobs:
...
       uses: actions/checkout@v3
-    - name: Test
-      run: make test-community
+    - name: Set up gotestsum
+      run: |
+        go install gotest.tools/gotestsum@latest
+        mkdir -p tmp/test-results
+    - name: Test 
+      run: |
+        CGO_ENABLED=1 gotestsum --junitfile tmp/test-results/test.xml --raw-command -- go test -json -tags=sources $(go list ./... | grep -v 'pkg/sources')
```
Comment (dustin-decker) [inline .github/workflows/test.yml:75]: Previously (and the make target) had `grep -v /vendor/ | grep -v pkg/detectors | grep -v pkg/sources)`. Why just the one exclusion now?

Comment (codevbus) [reply .github/workflows/test.yml]: Fixed; I had gotten the test make entry confused.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -66,5 +66,27 @@ jobs:
...
+      if: '!cancelled()' # Run this step even when the tests fail. Skip if the workflow is cancelled.
+      uses: buildpulse/buildpulse-action@main
+      with:
+        account: 79229934
+        repository: 77726177
+        path: |
+          tmp/test-results/*.xml
+        key: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
+        secret: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
+        tags: scanner
```
Comment (dustin-decker) [inline .github/workflows/test.yml:87]: Should we update this to community?

**Review by dustin-decker — APPROVED** [review]

**Review by dustin-decker — APPROVED** [review]

**Pull request merged**